packages/shared/src/hooks/useUpdateEmailNotificationSettings.ts

```diff
@@ -0,0 +1,246 @@
+      };
+
+      try {
+        const response = await updateEmailNotificationSettings({
+          unsubscribedFromNotifications: !settingsUpdate.notifications,
+          unsubscribedFromDigest: !settingsUpdate.digest,
+          unsubscribedFromMarketing: !settingsUpdate.marketing,
+          unsubscribedFromMembersClub: !settingsUpdate.membersClub ?? false,
+          unsubscribedFromAll: false,
+        });
+
+        if (
+          response.updateEmailNotificationSettings?.viewer?.email?.emailNotificationSettings &&
+          settingType in
+            response.updateEmailNotificationSettings.viewer.email.emailNotificationSettings
+        ) {
+          const settings =
+            response.updateEmailNotificationSettings.viewer.email.emailNotificationSettings;
+          if (settings[settingType as keyof EmailSettings] === newSettingValue) {
+            setEmailSettings(settingsUpdate);
+            pushToast({
+              message: `Settings successfully updated. You will ${
+                settingsUpdate[settingType] ? 'no longer' : 'now'
+              } receive ${settingTitle} emails.`,
+            });
+          }
+        }
+      } catch (error) {
+        if (error instanceof Error) {
+          reportError('Failed to update email notification settings');
+        }
+        pushToast({
+          message: 'Unfortunately, there was an error updating your notification settings.',
+        });
+      }
+    },
+    [emailSettings, updateEmailNotificationSettings]
+  );
+
+  const handleToggle = useCallback(
+    async ({ settingType, settingTitle, pushToast }: handleToggleProps) => {
+      const newSettingValue = !emailSettings[settingType];
+
+      setEmailSettings((prevSettings) => ({
+        ...prevSettings,
+        [settingType]: newSettingValue,
+      }));
+
+      try {
+        await handleEmailNotificationChange({
+          settingType,
+          newSettingValue,
+          settingTitle,
+          pushToast,
+        });
+
+        pushToast({
+          message: `Settings successfully updated. You have ${
+            newSettingValue ? 'subscribed to' : 'unsubscribed from'
+          } ${settingTitle}.`,
+        });
+      } catch (error) {
+        setEmailSettings((prevSettings) => ({
+          ...prevSettings,
+          [settingType]: !newSettingValue,
+        }));
+
+        reportError('Failed to update email notification settings');
+
+        pushToast({
+          message: 'Unfortunately, there was an error updating your notification settings.',
+        });
+      }
+    },
+    [emailSettings, handleEmailNotificationChange]
+  );
+
+  return useMemo(
+    () => ({
+      emailSettings,
+      computeToggleChecked,
+      handleToggle,
+    }),
+    [emailSettings, computeToggleChecked, handleToggle]
+  );
+}
+
+type EmailNotificationSettings = {
+  notifications: boolean;
+  marketing: boolean;
+  digest: boolean;
+  membersClub: boolean;
+  all: boolean;
+  [key: string]: boolean;
```

Comment:
ah i see what you're saying, it's fine we can fix this, will comment separately :)